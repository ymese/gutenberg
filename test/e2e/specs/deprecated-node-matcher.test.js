/**
 * Internal dependencies
 */
import {
	newPost,
	insertBlock,
	getEditedPostContent,
} from '../support/utils';
import { activatePlugin, deactivatePlugin } from '../support/plugins';

describe( 'Deprecated Node Matcher', () => {
	beforeAll( async () => {
		await activatePlugin( 'gutenberg-test-deprecated-node-matcher' );
	} );

	beforeEach( async () => {
		await newPost();
	} );

	afterAll( async () => {
		await deactivatePlugin( 'gutenberg-test-deprecated-node-matcher' );
	} );

	it( 'should insert block', async () => {
		await insertBlock( 'Deprecated Node Matcher' );
		await page.keyboard.type( 'test' );
		await page.keyboard.press( 'Enter' );
		expect( console ).toHaveWarned();
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
