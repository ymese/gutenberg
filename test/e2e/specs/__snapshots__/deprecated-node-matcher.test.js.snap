// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deprecated Node Matcher should insert block 1`] = `
"<!-- wp:deprecated-node-matcher -->
<blockquote class=\\"wp-block-deprecated-node-matcher\\"><p>test</p><p></p></blockquote>
<!-- /wp:deprecated-node-matcher -->"
`;
